firmware/arduino/chessboard/uart.ino

```diff
@@ -23,70 +52,79 @@ void serialEvent2()
         }
 
         // Check if the buffer is full, process input
-        if (byteNum  ==  5)
+        if (byteNum  ==  6)
         {
             // Reset buffer position
             byteNum = -1;
 
             // Process input
             // Returns true for valid input and false for invalid input. Calls movement function
-            parseMessageFromPi(buffer);
+            currentState = EXECUTING;
+            if (parseMessageFromPi(buffer))
+                errorCode = NONE;
+
+            // Send response
+            sendMessageToPi(currentState, buffer[5], errorCode);
         }
     }
 }
 
 bool parseMessageFromPi(char * message)
```

Comment:
- To simplify the amount of nested if statements/duplicated code for validation, you can validate the input at the beginning of the function. What you have is good, but just an alternate approach to separate validation from the other logic stuff
```
if (message[0] == ALIGN) {
    // validate start
    // set startPoint
}
if (message[0] == DIRECT || message[0] == EDGES) {
    // validate start and end
    // set startPoint, endPoint
}

if (message[0] == DIRECT) {
    // blah
} else if (message[0] == EDGES) {
    // blah
} else if (message[0] == ALIGN) {
    // blah
} else {
    // blah
}

// blah
```